deployment/initialize_keycloak.sh

```diff
@@ -39,6 +50,19 @@ timeout 300 bash -c "while ! ssh ubuntu@\"$target_server_url\" \"cd $location &&
                        sleep 5;
                      done"
 
+if ssh ubuntu@"$target_server_url" "ls $keycloak_user_dir/*-users-*.json &>/dev/null"; then
+  echo "Testing if the number of current users matches the number of exported users"
+  current_users=$(ssh ubuntu@"$target_server_url" "docker exec $container_name /opt/keycloak/bin/kcadm.sh get users -r datalandsecurity --server http://localhost:8080/keycloak --realm master --user $KEYCLOAK_ADMIN --password $KEYCLOAK_ADMIN_PASSWORD | grep -c '\"username\" :'")
```

Comment:
What are the current users / how are they different from all users?
If current users = all users \ test users, why do we need the all users as separate entitity?

Reply:
This is actually a test that can only be executed at this point of the keycloak initialization. We are comparing the number of users that are currently stored in the keycloak docker volume (current_users) vs the number of users that should have been imported (expected_users=all_users-test_users) where test_users is the number of users generated by E2E tests and therefore should and are not imported again.